packages/host/app/services/matrix-service.ts

```diff
@@ -217,11 +223,21 @@ export default class MatrixService extends Service {
       await this.flushMembership;
       await this.flushTimeline;
       clearAuth();
-      this.realm.logout();
-      this.realmServer.logout();
-      this.cardService.resetState();
+      this.reset.resetAll();
       this.unbindEventListeners();
       await this.client.logout(true);
+      // when user logs out we transition them back to an empty stack with the
+      // workspace chooser open. this was we don't inadvertently leak private
+      // card id's in the URL
```

Comment:
Do note that I had to make an executive decision around how to handle URL state when the user logs out. I opted to empty the stack and show the workspace chooser, so that the next user wouldn't get exposure to potentially private realm artifacts that were serialized in the URL